core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -357,6 +361,26 @@ static Schema copyRecord(Schema record, List<Schema.Field> newFields, String new
     return copy;
   }
 
+  static Schema copyArray(Schema array, Schema elementSchema) {
+    Preconditions.checkArgument(array.getType() == org.apache.avro.Schema.Type.ARRAY,
+        "Cannot invoke copyArray on non array schema");
+    Schema copy = Schema.createArray(elementSchema);
+    for (Map.Entry<String, Object> prop : array.getObjectProps().entrySet()) {
+      copy.addProp(prop.getKey(), prop.getValue());
+    }
+    return copy;
+  }
+
+  static Schema copyMap(Schema map, Schema valueSchema) {
+    Preconditions.checkArgument(map.getType() == org.apache.avro.Schema.Type.MAP,
+        "Cannot invoke copyMap on non map schema");
```

Comment:
I think this should include the problem schema for debugging.

Reply:
Added